libs/libc/semaphore/sem_trywait.c

```diff
@@ -127,8 +128,9 @@ int nxsem_trywait(FAR sem_t *sem)
 #endif
       )
     {
-      int32_t old = 1;
-      return atomic_try_cmpxchg_acquire(NXSEM_COUNT(sem), &old, 0) ?
+      int32_t old = NXSEM_NO_MHOLDER;
+      return atomic_try_cmpxchg_acquire(NXSEM_MHOLDER(sem), &old,
```

Comment:
No reason to keep re-reading GETTID() over and over, it will not change. Take a local copy.

Reply:
done, thanks